src/lib.rs

```diff
@@ -29,296 +29,70 @@
-#[derive(Clone, Debug, RustcDecodable, RustcEncodable)]
-pub struct MpidMessage {
-    /// Header part of the message
-    header: MpidHeader,
-    /// Receiver's identification
-    recipient: XorName,
-    /// Content part of the message
-    body: Vec<u8>,
-    /// Signature of the message
-    signature: Signature,
-}
-
-impl MpidMessage {
-    /// Constructor for MpidMessage
-    pub fn new(header: MpidHeader,
-               recipient: XorName,
-               body: Vec<u8>,
-               secret_key: &SecretKey)
-               -> Option<MpidMessage> {
-        if body.len() > MAX_BODY_SIZE {
-            return None;
-        }
+/// Maximum allowed inbox size for an account (128 MiB).
+pub const MAX_INBOX_SIZE: usize = 1 << 27;
+/// Maximum allowed outbox size for an account (128 MiB).
+pub const MAX_OUTBOX_SIZE: usize = 1 << 27;
 
-        let recipient_and_body = Self::encode(&recipient, &body);
-        Some(MpidMessage {
-            header: header,
-            recipient: recipient,
-            body: body,
-            signature: sign::sign_detached(&recipient_and_body, secret_key),
-        })
-    }
-
-    /// Getter for MpidMessage::header
-    pub fn header(&self) -> &MpidHeader {
-        &self.header
-    }
-
-    /// Getter for MpidMessage::recipient
-    pub fn recipient(&self) -> &XorName {
-        &self.recipient
-    }
+mod error;
+mod mpid_header;
+mod mpid_message;
+mod mpid_message_wrapper;
 
-    /// Getter for MpidMessage::body
-    pub fn body(&self) -> &Vec<u8> {
-        &self.body
-    }
+pub use error::Error;
+pub use mpid_header::{MpidHeader, MAX_HEADER_METADATA_SIZE};
+pub use mpid_message::{MpidMessage, MAX_BODY_SIZE};
+pub use mpid_message_wrapper::MpidMessageWrapper;
 
-    /// Help function to verify whether a MpidMessage is valid
-    pub fn verify(&self, public_key: &PublicKey) -> bool {
-        let encoded = Self::encode(&self.recipient, &self.body);
-        sign::verify_detached(&self.signature, &encoded, public_key) &&
-            self.header.verify(public_key)
-    }
+use std::fmt::Write;
 
-    fn encode(recipient: &XorName, body: &Vec<u8>) -> Vec<u8> {
-        let message_for_sign = MessageForSign {
-            recipient: recipient.clone(),
-            body: body.clone(),
-        };
-        match serialise(&message_for_sign) {
-            Ok(encoded) => return encoded,
-            Err(error) => {
-                error!("Failed to encode MpidMessage: {:?}", error);
-                return vec![];
-            }
-        }
-    }
-}
-
-/// Help function to calculate the name of a MpidHeader
-pub fn mpid_header_name(mpid_header: &MpidHeader) -> Option<XorName> {
-    match serialise(&mpid_header) {
-        Ok(encoded) => return Some(XorName(sha512::hash(&encoded[..]).0)),
-        Err(error) => {
-            error!("Failed to serialise Put request: {:?}", error);
+// Format a vector of bytes as a hexadecimal number, ellipsising all but the first and last three.
```

Comment:
Stolen from https://github.com/Fraser999/routing/blob/master/src/utils.rs#L20 unfortunately.  This should probably be a function in the maidsafe_utilities crate to be honest.